reactor-core/src/main/java/reactor/core/publisher/Operators.java

```diff
@@ -1512,30 +1532,30 @@ public final boolean set(Subscription s) {
 				return false;
 			}
 
-			if (S.compareAndSet(this, null, s)) {
+			long r;
+			long accomulated = 0;
+			for (;;) {
+				r = this.requested;
 
-				long r = REQUESTED.getAndSet(this, 0L);
-
-				if (r != 0L) {
-					s.request(r);
+				if (r == STATE_CANCELLED || r == STATE_SUBSCRIBED) {
+					s.cancel();
+					return false;
 				}
 
-				return true;
-			}
+				this.s = s;
 
-			a = this.s;
+				long toRequest = r - accomulated;
+				if (toRequest > 0) { // if there is something,
+					s.request(toRequest); // then we do a request on the given subscription
+				}
+				accomulated = r;
```

Comment:
suggestion: `accumulated += toRequest`. it simplifies to the same, but is easier to read and better conveys the logic IMHO (we track how much of the pre-subscribe request we've already propagated)

Reply:
is not it redundant?

since we calculate `dif` and that add this `dif` back

Reply:
I'm just saying that we are doing 

substruction from r (`long diff = r - accumulated`) and addition to accumulated that `diff`, so it is equal to `r` in any, case

Reply:
yeah, that's what I meant by "it simplifies to the same". but I find the logic far easier to follow in this form. it makes it more visible that `accumulated` accumulates the consecutive demands that have already been propagated. just "cosmetic"

Reply:
alright. Np

Reply:
fixed